### PR TITLE
Create python-publish.yml

### PR DESCRIPTION
**Pull request by vshampor**
Will publish a package to PyPI upon a Github release.

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This workflows will upload a Python Package using Twine when a release is created
```
Comment (AlexKoff88) [inline .github/workflows/python-publish.yml:1]: What about the licensing of such files?

Comment (vshampor) [reply .github/workflows/python-publish.yml:1]: No idea. I just used a template file as-is, available from https://github.com/openvinotoolkit/nncf_pytorch/actions. Should we just slap an Apache header onto this file?

Comment (AlexKoff88) [reply .github/workflows/python-publish.yml:1]: Probably yes, I actually meant this.

Comment (vshampor) [reply .github/workflows/python-publish.yml]: Done

**Pull request merged**